NewForce-Cohort-6/tabloidfullstack-mac-cheese
# TS | Deactivate/Activate profile | #25

This code addresses ticket #25 (deactivate user profiles)

https://trello.com/b/0CrteZkv/mac-n-cheese

Given an admin is vising the list of User Profiles
When they select the Deactivate option
Then they should be prompted to confirm deactivation

Given the admin wishes to confirm the deactivation
When they the select the option to confirm
Then the User Profile should be deactivated in the system
And the admin should be directed back to the User Profile list

Given the admin decides not to confirm the deactivation
When they the select the option to reject confirmation
Then the User Profile should NOT be deactivated
And the admin should be directed back to the User Profile list

Given a user is deactivated
When they try to login to the system
Then the system should behave as if the user does not exist
_____________________________________________________________
# Testing Instructions

* In your terminal, from your current branch:
-git add and git commit whatever you're working on in your branch
-git checkout main
-git fetch --a or git fetch all
-(if files do not show, git pull origin)
-git checkout DeactivateProfile-TS
-start the application with "start <application name>"

* Ensure that you have created the Tabloid database
* If needed:

-Use this to fill out tables and columns: 

https://github.com/NewForce-Cohort-6/tabloidfullstack-mac-cheese/blob/main/SQL/01_Tabloid_Create_DB.sql

-Use this to seed the database: 

https://github.com/NewForce-Cohort-6/tabloidfullstack-mac-cheese/blob/main/SQL/02_Tabloid_Seed_Data.sql

** You must run a couple sql commands on the Tabloid database to create the "isActive" property on userProfiles.
Run this SQLQuery:
    ALTER TABLE UserProfile ADD IsActive bit;

Next, you must update one of the UserProfiles for testing.
Run this SQLQuery:
    UPDATE UserProfile SET IsActive = 1 WHERE Id = 1;

* Check that the correct row updated and remember the UserProfile information for testing

* Launch the api from VisualStudio using the filled in green 'play' button at the top
* Lauch the front end code by right clicking the 'client' folder in visual studio solution explorer and selecting 'Open in terminal'
* Once the terminal opens, type 'code .'
* This should open visual studio code. Open a new terminal and type 'npm start'
* In the browser, if not logged in already, register an email with the registration button, then log in with that email
* Once logged in, select the "Users" tab from the navbar
* Once the users tab is selected, you should see a list of users with their display name, full name, and user type along with a button that says either "activate" or "deactivate" depending on the user's active status
* The user with an id of 1 should have an option to deactivate (ie, the user should be active) 
* Select user id =1 to deactivate. Once deactivated, the button should switch to "activate" to reactivate the user. 
* Check in the dev tools or database to confirm that the user's isActive property is changed to false.
* Select the activate button to reactivate.
* Check in the dev tools or database to confirm that the user's isActive property is changed to true.
* Once again, deactivate the user.
* Logout of the app and attempt to login as the user you previously deactivated. 
* You should not be allowed to login as the deactivated user.


___________
# Checklist:
